tests/unit/lms/services/canvas_test.py

```diff
@@ -55,8 +52,8 @@ def test_if_the_file_isnt_in_the_course_it_finds_a_matching_file_instead(
         assignment,
         public_url_for_file,
     ):
-        canvas_file_finder.assert_file_in_course.side_effect = (
-            CanvasFileNotFoundInCourse(sentinel.file_id)
+        canvas_file_finder.assert_file_in_course.side_effect = FileNotFoundInCourse(
+            "canvas_file_not_found_in_course", sentinel.file_id
         )
```

Comment:
Having to instantiate a full exception here kind of sucks. You have to do it because we have non-nullable inputs to the exception. I think it's not nice for a couple of reasons

#### It's not as neat
 
This is partially just general neat freakery on my part, but it also harms readability.
    
#### It includes spuriously accurate information

There are values in the test here which actually don't make any difference to the test. I think we should take a kind of "Chekhovs value" approach, that:

 * If it's here we should use it
 * If we don't use it it shouldn't be here

This is because it loads the test with lots of details which _might_ be relevant, but actually aren't. For example we use `sentinel.file_id` later on. Is that read from the exception? It's not, and if it wasn't here we can't make that incorrect leap. This also narrows the focus of what this test is, and isn't about. 

I think there's a competing school of thought that says values should be "realistic". I think the idea is to highlight how the code operates by reading the tests. 

#### What can we do about it?

a) We could **change the exception to have default values**, but in this case there aren't any defaults, and not providing inputs would be an error, so that's not great.

b) We could **choose names to convey meaning** and explain they don't matter.

```python
FileNotFoundInCourse("arbitrary_error_code", file_id="not_important")
```

c) We could **add a factory class** for these? These could re-used between tests, and a set of exception factories seems reasonable enough. Factories allow us to have randomised values, which positively states we don't care about the value. This is the opposite of having an irrelevant but likely looking string:

 * A fixed string: Implies the value is important
 * No value: Implies the value isn't important (or maybe the default is?)
 * A randomized value: States the value isn't important

d) **Do nothing**, as it's not super important, I mostly wanted to have a chat about testing methodology.

Reply:
Created an issue off this comment. I do revisit these as I add them to the project itself.